### PR TITLE
Move include/text.php to src/Content/Text

### PR DESCRIPTION
**Pull request by zeroadam**
Move include/text.php to new Text class in src/Content

relates to #3878

### CODE REVIEW

**Comment by MrPetovan** [comment]
> +4,340 −4,243 

*gulp*

**Comment by zeroadam** [comment]
@MrPetovan I apologized for this in the issue itself! Perhaps I should copy that here! 😁

File: src/Content/Text.php [hunk 1]
```diff
@@ -0,0 +1,1714 @@
...
+        return str_replace(["<", ">"], ['[', ']'], $string);
+    }
+
+    /**
+     * use this on "body" or "content" input where angle chars shouldn't be removed,
+     * and allow them to be safely displayed.
+     * @param string $string
+     * @return string
+     */
+    function escapeTags($string) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 2]
```diff
@@ -0,0 +1,1714 @@
...
+        return htmlspecialchars($string, ENT_COMPAT, 'UTF-8', false);
+    }
+
+    /**
+     * generate a string that's random, but usually pronounceable.
+     * used to generate initial passwords
+     * @param int $len
+     * @return string
+     */
+    function autoName($len) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 3]
```diff
@@ -0,0 +1,1714 @@
...
+
+        return $word;
+    }
+
+    /**
+     * escape text ($str) for XML transport
+     * @param string $str
+     * @return string Escaped text.
+     */
+    function xmlify($str) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, this methods is specific to XML text, it makes more sense to me to move it to the `Friendica\Content\Text\XML` class.

File: src/Content/Text.php [hunk 4]
```diff
@@ -0,0 +1,1714 @@
...
+
+        return $buffer;
+    }
+
+    /**
+     * undo an xmlify
+     * @param string $s xml escaped text
+     * @return string unescaped text
+     */
+    function unxmlify($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 5]
```diff
@@ -0,0 +1,1714 @@
...
+        $ret = htmlspecialchars_decode($s, ENT_QUOTES);
+        return $ret;
+    }
+
+    /**
+     * Loader for infinite scrolling
+     * @return string html for loader
+     */
+    function scrollLoader() {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 6]
```diff
@@ -0,0 +1,1714 @@
...
+        ]);
+    }
+
+    /**
+     * Turn user/group ACLs stored as angle bracketed text into arrays
+     *
+     * @param string $s
+     * @return array
+     */
+    function expandAcl($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php (hunk 5 above)
Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, this method is heavily coupled with the `App` object through `replaceMacros`, I think it makes more sense to move it to the `App` class.

File: src/Content/Text.php [hunk 7]
```diff
@@ -0,0 +1,1714 @@
...
+            }
+        }
+        return $ret;
+    }
+
+    /**
+     * Wrap ACL elements in angle brackets for storage
+     * @param string $item
+     */
+    function sanitiseAcl(&$item) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Typo: Please rename to `sanitizeAcl`.

File: src/Content/Text.php [hunk 8]
```diff
@@ -0,0 +1,1714 @@
...
+    /**
+     * Convert an ACL array to a storable string
+     *
+     * Normally ACL permissions will be an array.
+     * We'll also allow a comma-separated string.
+     *
+     * @param string|array $p
+     * @return string
+     */
+    function perms2Str($p) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 9]
```diff
@@ -0,0 +1,1714 @@
...
+    }
+
+    /**
+     * load template $s
+     *
+     * @param string $s
+     * @param string $root
+     * @return string
+     */
+    function getMarkupTemplate($s, $root = '') {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, this method is heavily coupled with the `App` object, I think it makes more sense to move it to the `App` class.

File: src/Content/Text.php [hunk 10]
```diff
@@ -0,0 +1,1714 @@
...
+     *  might match 'notclass3' and a regex to do the job is
+     *  possible but a bit complicated.
+     *  pass the attribute string as $attr and the attribute you
+     *  are looking for as $s - returns true if found, otherwise false
+     *
+     * @param string $attr attribute value
+     * @param string $s string to search
+     * @return boolean True if found, False otherwise
+     */
+    function attributeContains($attr, $s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 11]
```diff
@@ -0,0 +1,1714 @@
...
+        $a = explode(' ', $attr);
+        return (count($a) && in_array($s,$a));
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a space after commas.

File: src/Content/Text.php [hunk 12]
```diff
@@ -0,0 +1,1714 @@
...
+     * LOGGER_TRACE
+     * LOGGER_DEBUG
+     * LOGGER_DATA
+     * LOGGER_ALL
+     *
+     * @global array $LOGGER_LEVELS
+     * @param string $msg
+     * @param int $level
+     */
+    function logger($msg, $level = LOGGER_INFO) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, yaddi yadda please move it to the `App` class.

File: src/Content/Text.php [hunk 13]
```diff
@@ -0,0 +1,1714 @@
...
+     * LOGGER_TRACE
+     * LOGGER_DEBUG
+     * LOGGER_DATA
+     * LOGGER_ALL
+     *
+     * @global array $LOGGER_LEVELS
+     * @param string $msg
+     * @param int $level
+     */
+    function dlogger($msg, $level = LOGGER_INFO) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, yaddi yadda please move it to the `App` class.

File: src/Content/Text.php [hunk 14]
```diff
@@ -0,0 +1,1714 @@
...
+    }
+
+    /**
+     * Compare activity uri. Knows about activity namespace.
+     *
+     * @param string $haystack
+     * @param string $needle
+     * @return boolean
+     */
+    function activityMatch($haystack, $needle) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 15]
```diff
@@ -0,0 +1,1714 @@
...
+     * We also get @person@domain.com - which would make
+     * the regex quite complicated as tags can also
+     * end a sentence. So we'll run through our results
+     * and strip the period from any tags which end with one.
+     * Returns array of tags found, or empty array.
+     *
+     * @param string $string Post content
+     * @return array List of tag and person names
+     */
+    function getTags($string) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 16]
```diff
@@ -0,0 +1,1714 @@
...
+        return $ret;
+    }
+
+    /**
+     * quick and dirty quoted_printable encoding
+     *
+     * @param string $s
+     * @return string
+     */
+    function qp($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 17]
```diff
@@ -0,0 +1,1714 @@
...
+    }
+
+    /**
+     * Get html for contact block.
+     *
+     * @template contact_block.tpl
+     * @hook contact_block_end (contacts=>array, output=>string)
+     * @return string
+     */
+    function contactBlock() {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, it looks like this should be move to a Module. Where is it used?

File: src/Content/Text.php [hunk 18]
```diff
@@ -0,0 +1,1714 @@
...
+    *	string 'thumb' => The contact picture
+    *	string 'click' => js code which is performed when clicking on the contact
+    * @param boolean $redirect If true try to use the redir url if it's possible
+    * @param string $class CSS class for the
+    * @param boolean $textmode If true display the contacts as text links
+    *	if false display the contacts as picture links
+
+    * @return string Formatted html
+    */
+    function micropro($contact, $redirect = false, $class = '', $textmode = false) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 19]
```diff
@@ -0,0 +1,1714 @@
...
+    {
+        return preg_match('/^[_a-zA-Z0-9\-\+]+(\.[_a-zA-Z0-9\-\+]+)*@[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)+$/', $email_address);
+    }
+
+    /**
+     * Replace naked text hyperlink with HTML formatted hyperlink
+     *
+     * @param string $s
+     */
+    function linkify($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 20]
```diff
@@ -0,0 +1,1714 @@
...
+    }
+
+    /**
+     * Load poke verbs
+     *
+     * @return array index is present tense verb
+     * 				 value is array containing past tense verb, translation of present, translation of past
+     * @hook poke_verbs pokes array
+     */
+    function getPokeVerbs() {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 21]
```diff
@@ -0,0 +1,1714 @@
...
+        return $arr;
+    }
+
+    /**
+     * @brief Translate days and months names.
+     *
+     * @param string $s String with day or month name.
+     * @return string Translated string.
+     */
+    function dayTranslate($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, I think it should be moved to the `L10n` class.

File: src/Content/Text.php [hunk 22]
```diff
@@ -0,0 +1,1714 @@
...
+        return $ret;
+    }
+
+    /**
+     * @brief Translate short days and months names.
+     *
+     * @param string $s String with short day or month name.
+     * @return string Translated string.
+     */
+    function dayShortTranslate($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, I think it should be moved to the `L10n` class.

File: src/Content/Text.php [hunk 23]
```diff
@@ -0,0 +1,1714 @@
...
+        return $ret;
+    }
+
+    /**
+     * Normalize url
+     *
+     * @param string $url
+     * @return string
+     */
+    function normaliseLink($url) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 24]
```diff
@@ -0,0 +1,1714 @@
...
+     * slight but hopefully insignificant differences such as if one
+     * is https and the other isn't, or if one is www.something and
+     * the other isn't - and also ignore case differences.
+     *
+     * @param string $a first url
+     * @param string $b second url
+     * @return boolean True if the URLs match, otherwise False
+     *
+     */
+    function linkCompare($a, $b) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 25]
```diff
@@ -0,0 +1,1714 @@
...
+        return $html;
+    }
+
+    /**
+     * @brief Given a text string, convert from bbcode to html and add smilie icons.
+     *
+     * @param string $text String with bbcode.
+     * @return string Formattet HTML.
+     */
+    function prepareText($text) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 26]
```diff
@@ -0,0 +1,1714 @@
...
+
+        return [$categories, $folders];
+    }
+
+    /**
+     * get private link for item
+     * @param array $item
+     * @return boolean|array False if item has not plink, otherwise array('href'=>plink url, 'title'=>translated title)
+     */
+    function getPlink($item) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 27]
```diff
@@ -0,0 +1,1714 @@
...
+
+        return $ret;
+    }
+
+    /**
+     * replace html amp entity with amp char
+     * @param string $s
+     * @return string
+     */
+    function unamp($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 28]
```diff
@@ -0,0 +1,1714 @@
...
+        return str_replace('&amp;', '&', $s);
+    }
+
+
+    /**
+     * return number of bytes in size (K, M, G)
+     * @param string $size_str
+     * @return number
+     */
+    function returnBytes($size_str) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 29]
```diff
@@ -0,0 +1,1714 @@
...
+            default: return $size_str;
+        }
+    }
+
+    /**
+     * @param string $s
+     * @param boolean $strip_padding
+     * @return string
+     */
+    function base64UrlEncode($s, $strip_padding = false) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 30]
```diff
@@ -0,0 +1,1714 @@
...
+        }
+
+        return $s;
+    }
+
+    /**
+     * @param string $s
+     * @return string
+     */
+    function base64UrlDecode($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 31]
```diff
@@ -0,0 +1,1714 @@
...
+
+        return base64_decode(strtr($s,'-_','+/'));
+    }
+
+    /**
+     * return div element with class 'clear'
+     * @return string
+     * @deprecated
+     */
+    function clearDiv() {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 32]
```diff
@@ -0,0 +1,1714 @@
...
+        return '<div class="clear"></div>';
+    }
+
+    function bbTranslateVideo($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 33]
```diff
@@ -0,0 +1,1714 @@
...
+                    $s = str_replace($mtch[0], '[youtube]' . $mtch[1] . '[/youtube]', $s);
+                } elseif (stristr($mtch[1], 'vimeo')) {
+                    $s = str_replace($mtch[0], '[vimeo]' . $mtch[1] . '[/vimeo]', $s);
+                }
+            }
+        }
+        return $s;
+    }
+
+    function HTML2BBVideo($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, this method should be moved to the `Friendica\Content\text\HTML` class.

File: src/Content/Text.php [hunk 34]
```diff
@@ -0,0 +1,1714 @@
...
+    
+        return $s;
+    }
+
+    /**
+     * apply xmlify() to all values of array $val, recursively
+     * @param array $val
+     * @return array
+     */
+    function arrayXmlify($val){
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, this method should be moved to the `Friendica\Content\Text\XML` class.

File: src/Content/Text.php [hunk 35]
```diff
@@ -0,0 +1,1714 @@
...
+    }
+
+    /**
+     * transform link href and img src from relative to absolute
+     *
+     * @param string $text
+     * @param string $base base url
+     * @return string
+     */
+    function relToAbs($text, $base) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 36]
```diff
@@ -0,0 +1,1714 @@
...
+        return $text;
+    }
+
+    /**
+     * get translated item type
+     *
+     * @param array $itme
+     * @return string
+     */
+    function itemPostType($item) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

Comment (MrPetovan) [reply src/Content/Text.php]: Additionally, it should be moved to the `Friendica\Model\Item` class.

File: src/Content/Text.php [hunk 37]
```diff
@@ -0,0 +1,1714 @@
...
+
+        return L10n::t('post');
+    }
+
+    // post categories and "save to file" use the same item.file table for storage.
+    // We will differentiate the different uses by wrapping categories in angle brackets
+    // and save to file categories in square brackets.
+    // To do this we need to escape these characters if they appear in our tag.
+
+    function fileTagEncode($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 38]
```diff
@@ -0,0 +1,1714 @@
...
+        return str_replace(['<','>','[',']'],['%3c','%3e','%5b','%5d'],$s);
+    }
+
+    function fileTagDecode($s) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 39]
```diff
@@ -0,0 +1,1714 @@
...
+        return str_replace(['%3c', '%3e', '%5b', '%5d'], ['<', '>', '[', ']'], $s);
+    }
+
+    function fileTagFileQuery($table,$s,$type = 'file') {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards:
- Please add a new line before class method opening brace.
- Please add a space after commas.

File: src/Content/Text.php [hunk 40]
```diff
@@ -0,0 +1,1714 @@
...
+        if ($type == 'file') {
+            $str = preg_quote('[' . str_replace('%', '%%', self::fileTagEncode($s)) . ']');
+        } else {
+            $str = preg_quote('<' . str_replace('%', '%%', self::fileTagEncode($s)) . '>');
+        }
+        return " AND " . (($table) ? DBA::escape($table) . '.' : '') . "file regexp '" . DBA::escape($str) . "' ";
+    }
+
+    // ex. given music,video return <music><video> or [music][video]
+    function fileTagListToFile($list, $type = 'file') {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 41]
```diff
@@ -0,0 +1,1714 @@
...
+                if (strlen($item)) {
+                    $tag_list .= $lbracket . self::fileTagEncode(trim($item))  . $rbracket;
+                }
+            }
+        }
+        return $tag_list;
+    }
+
+    // ex. given <music><video>[friends], return music,video or friends
+    function fileTagFileToList($file, $type = 'file') {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 42]
```diff
@@ -0,0 +1,1714 @@
...
+                    $list .= ',';
+                }
+                $list .= self::fileTagDecode($mtch[1]);
+            }
+        }
+
+        return $list;
+    }
+
+    function fileTagUpdatePconfig($uid, $file_old, $file_new, $type = 'file') {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 43]
```diff
@@ -0,0 +1,1714 @@
...
+            }
+        }
+    
+        return false;
+    }
+    
+    /**
+     * remove intentation from a text
+     */
+    function deindent($text, $chr = "[\t ]", $count = NULL) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 44]
```diff
@@ -0,0 +1,1714 @@
...
+        }
+    
+        for ($k = 0; $k < count($lines); $k++) {
+            $lines[$k] = preg_replace("|^" . $chr . "{" . $count . "}|", "", $lines[$k]);
+        }
+    
+        return implode("\n", $lines);
+    }
+    
+    function formatBytes($bytes, $precision = 2) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

File: src/Content/Text.php [hunk 45]
```diff
@@ -0,0 +1,1714 @@
...
+    /**
+     * @brief translate and format the networkname of a contact
+     *
+     * @param string $network
+     *	Networkname of the contact (e.g. dfrn, rss and so on)
+     * @param sting $url
+     *	The contact url
+     * @return string
+     */
+    function formatNetworkName($network, $url = 0) {
```
Comment (MrPetovan) [inline src/Content/Text.php]: Code standards: Please add a new line before class method opening brace.

**Comment by MrPetovan** [comment]
Please review all my comments that have been marked as "outdated"  after you added function scope. You probably should mention "static" as well by the way.

**Comment by annando** [comment]
BTW: @zeroadam nice having you back!

**Comment by MrPetovan** [comment]
Better suggestion: Start extracting a couple functions from `include/text.php`at a time. You can stash your current work and pop it later not to have to redo everything.

Suggested plan (one PR per line):
1. Extract `logger()` and `dlogger()` to a new `Friendica\Core\Logger` static class.
2. Extract `replace_macro()` and `get_markup_template()` to a new `Friendica\Core\Renderer` static class.
3. Extract XML and HTML-related functions (`xmlify`, `array_xmlify`, `unamp`, `unxmlify`, `html2bb_video`, ...) to `Friendica\Content\Text\XML` and `Friendica\Content\Text\HTML`.
4. Extract l10n-related functions to `Friendica\Core\L10n`.
5. Extract Item-related functions (`put_item_in_cache`, `prepare_body`, `item_post_type`, ...) to `Friendica\Model\Item`.
6. Extract "File Tag" functions (`file_tag_*`, ...) to a new `Friendica\Model\FileTag` class.

**Comment by zeroadam** [comment]
Copied comments/notes, and am going to kill this PR, as well as addons. Will work on the latest set of class moves/new classes and submit. That way i'm not pushing a bunch of testing commits

**Comment by annando** [comment]
@MrPetovan Whatever there will be done: We should ensure to be able to make quick PRs. Since so many files will be touched, it will be better to have a short time between creating of the PR and accepting it. This will create fewer conflicts with parallel work.

**Comment by zeroadam** [comment]
Closing in favor of smaller PRs...

**Comment by MrPetovan** [comment]
@annando Smaller PRs: easier to accept and merge. There will be some conflicts, inevitably, but splitting this PR in many smaller will reduce them to a minimum.

**Comment by annando** [comment]
@MrPetovan That's what I mean.

**Pull request closed without merging**